File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/ProductDao.java
package by.bstu.vs.stpms.courier_application.model.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import by.bstu.vs.stpms.courier_application.model.database.entity.Product;

@Dao
public abstract class ProductDao extends AbstractDao<Product> {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract void insertAll(List<Product> products);

    @Override
    @Query("SELECT * FROM product WHERE id = :id")
    public abstract Product findById(long id);
    @Override
    @Query("SELECT * FROM product")
    public abstract List<Product> getAll();
}

File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/ProductDao.kt
package by.bstu.vs.stpms.courier_application.model.database.dao

import androidx.room.*
import by.bstu.vs.stpms.courier_application.model.database.entity.Product

@Dao
abstract class ProductDao : AbstractDao<Product>() {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insertAll(products: List<Product>)
    @Query("SELECT * FROM product WHERE id = :id")
    abstract override suspend fun findById(id: Long): Product?

    @Query("SELECT * FROM product")
    abstract override suspend fun getAll(): List<Product>
}